node/rpcstack.go

```diff
@@ -175,11 +173,20 @@ func (h *httpServer) start() error {
 }
 
 func (h *httpServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// check if ws request if ws enabled
+	ws := h.wsHandler.Load().(*rpcHandler)
+	if ws != nil && isWebsocket(r) {
+		if r.RequestURI == h.wsConfig.path {
+			ws.ServeHTTP(w, r)
+			return
+		}
+	}
+
 	rpc := h.httpHandler.Load().(*rpcHandler)
 	if rpc != nil {
 		// if request on root path and http-rpc handler is mounted
 		// on root path, serve request.
-		if r.RequestURI == "/" && h.httpPath == "/" {
+		if r.RequestURI == "/" && h.httpConfig.path == "/" {
```

Comment:
I don't really understand why this is only for `/` path. Can't we check for the configure path here and be done with the whole path thing?

Reply:
I guess, but then I wouldn't be using the mux for anything besides graphql requests which seems odd. The mux should probably route any handlers below root, no? The reason I had to include this weird condition is because if a handler is registered on root with the mux, it'll be served on all paths for some reason. That's why I wanted to use `gorilla/mux` package because this is not the case with their router.

Regardless, I have to do this pattern-matching manually for websocket requests anyway, so maybe I could do the same for http-rpc.

Reply:
> I guess, but then I wouldn't be using the mux for anything besides graphql requests 

I think it's OK not to use the mux for everything.

Reply:
Also FWIW I believe as currently coded this has the same issue as recent versions of geth 1.9.x where `/?foo=bar` isn't handled though it probably should be.

Reply:
Hey @ryanschneider , you need support for path prefixes then, right? I'll add that in now. I believe @fjl signed off on that too. https://github.com/ethereum/go-ethereum/pull/21826#issuecomment-747120439

Thanks for reminding me and checking out the changes.

Reply:
@renaynay yes please!  Thanks for doing this work and keeping our needs in mind when doing so.